plugins/in_head/in_head.c

```diff
@@ -426,7 +426,7 @@ static struct flb_config_map config_map[] = {
      "Set the record key"
     },
     {
-      FLB_CONFIG_MAP_INT, "buf_size", DEFAULT_BUF_SIZE,
+      FLB_CONFIG_MAP_SIZE, "buf_size", DEFAULT_BUF_SIZE,
```

Comment:
This seems like an ergonomic change that makes sense but is out of context in this PR which is meant to address a lower level issue so I'd like to see it moved to a different PR to make it easier to track the origin of changes and keep things clean.

Reply:
@leonardo-albertovich , this is also an endian issue, [buf_size](https://github.com/fluent/fluent-bit/blob/master/plugins/in_head/in_head.h#L35) is of type size_t inside the `flb_in_head_config` struct and takes 8 bytes, using FLB_CONFIG_MAP_INT here will lead to endian issue since it is read as 4 bytes

Reply:
Endianness only dictates how the components of a word are laid out in memory, not the size of the word. 

This is an error, but the error is that the wrong data type was used and thus either making `buf_size` an integer to match the definition OR changing the definition to match the type of `buf_size` are correct with the later being preferable because it improves the user experience.

So to be clear, it's an error but byte order is only a sign, not the source.